node/src/Pos/Block/Network/Retrieval.hs

```diff
@@ -305,17 +306,20 @@ handleCHsValid enqueue endedRecoveryVar nodeId lcaChild newestHash = do
                 -- difficulty than ncrecoveryheader, we're
                 -- gracefully exiting recovery mode.
                 let isMoreDifficultThan b x = b ^. difficultyL >= x ^. difficultyL
-                endedRecovery <- atomically $ do
+                atomically $ do
                     mRecHeader <- tryReadTMVar recHeaderVar
                     case mRecHeader of
                         Nothing -> return False
                         Just (_, rHeader) ->
                             if any (`isMoreDifficultThan` rHeader) blocks
                                 then isJust <$> tryTakeTMVar recHeaderVar
                                 else return False
-                void $ tryPutMVar endedRecoveryVar endedRecovery
-    void $ waitForConversations its
+    results <- waitForConversations $ fmap try convs
```

Comment:
I don't quite understand what's going on with `Any` here. We're creating a single conversation here, so it will have a single result, right? What are the things that we combine?